packages/components/src/components/buttons/Button/Button.tsx

```diff
@@ -131,7 +131,7 @@ export const Button = ({
         />
     ) : null;
 
-    const Loader = <Spinner size={getIconSize(size)} />;
+    const Loader = <Spinner size={getIconSize(size)} dataTest={rest['data-test']} />;
```

Comment:
This seems to me like we will have duplicit test id or inner component is too smart. 
In both cases it's not predictable what exactly is happening. 

I would suggest use it in `Spinner` like this:

```tsx
data-test={dataTest ? dataTest : `@spinner`}
```

instead of:

```tsx
data-test={dataTest ? `${dataTest}/spinner` : `@spinner`}
```


And of course it needs to specify it here:
```tsx
    const Loader = <Spinner size={getIconSize(size)} dataTest={`${rest['data-test']}/spinner`} />;
```

WDYT?

Reply:
this does not allow me to pass a distinguishable data-test to a spinner that is part of Button component. Both Button and Spinner rendered by the Button would have the same data-test

Reply:
sounds right, fixed up